src/scala/org/pantsbuild/zinc/bootstrapper/Cli.scala

```diff
@@ -0,0 +1,69 @@
+package org.pantsbuild.zinc.bootstrapper
+
+import java.io.File
+
+case class Configuration(
+  outputPath: File = new File("."),
+  compilerInterface: File = new File("."),
+  compilerBridgeSource: File = new File("."),
+  scalaCompiler: File = new File("."),
+  scalaLibrary: File = new File("."),
+  scalaReflect: File = new File(".")
+)
+
+object Cli {
+  val CliParser = new scopt.OptionParser[Configuration]("scopt") {
+    head("zinc-boostrapper", "0.0.1")
+
+    opt[File]('o', "out")
+      .required()
+      .valueName("<file>")
+      .action((x, c) => c.copy(outputPath = x))
+      .text("Output path for the compiler-bridge jar.")
+
+    opt[File]("compiler-interface")
+      .required()
+      .valueName("<file>")
+      .action((x, c) => c.copy(compilerInterface = x))
+      .validate { file =>
+        if (file.exists) success else failure(s"$file does not exist.")
+      }
+      .text("Compiler interface jar.")
+
+    opt[File]("compiler-bridge-src")
+      .required()
+      .valueName("<file>")
+      .action((x, c) => c.copy(compilerBridgeSource = x))
+      .validate { file =>
+        if (file.exists) success else failure(s"$file does not exist.")
+      }
+      .text("Compiler bridge source code.")
+
+    opt[File]("scala-compiler")
+      .required()
+      .valueName("<file>")
+      .action((x, c) => c.copy(scalaCompiler = x))
+      .validate { file =>
+        if (file.exists) success else failure(s"$file does not exist.")
+      }
+      .text("Path to the scala compiler.")
+
+    opt[File]("scala-library")
+      .required()
+      .valueName("<file>")
+      .action((x, c) => c.copy(scalaLibrary = x))
+      .validate { file =>
+        if (file.exists) success else failure(s"$file does not exist.")
+      }
+      .text("Path to the scala rumtime library.")
+
+    opt[File]("scala-reflect")
+      .required()
+      .valueName("<file>")
+      .action((x, c) => c.copy(scalaReflect = x))
+      .validate { file =>
+        if (file.exists) success else failure(s"$file does not exist.")
+      }
+      .text("Path to the scala reflection library.")
```

Comment:
Wow. scopt is really slick!

Reply:
Yep :) It's unfortunate, because writing a CLI parser library is a thing I've wanted to do for a long time now, but with things like scopt and clap (for Rust), there's never a good reason to!